### PR TITLE
feat: Add missing tket2 quantum gates, generalise RotationCompile

### PR DESCRIPTION
**Pull request by jake-arkinstall**
Addresses #484

### CODE REVIEW

File: guppylang/prelude/_internal/compiler/quantum.py [hunk 1]
```diff
@@ -81,11 +81,11 @@ def compile_with_inouts(self, args: list[Wire]) -> CallReturnWires:
...
             quantum_op(self.opname)(
-                ht.FunctionType([ht.Qubit, ROTATION_T], [ht.Qubit]), []
+                ht.FunctionType([ht.Qubit for _ in qs] + [ROTATION_T], [ht.Qubit for _ in qs]), []
             ),
-            q,
+            *qs,
             conditional,
         )
-        return CallReturnWires(regular_returns=[], inout_returns=[q])
+        return CallReturnWires(regular_returns=[], inout_returns=qs)
```
Comment (mark-koch) [inline guppylang/prelude/_internal/compiler/quantum.py:91]:
```suggestion
        return CallReturnWires(regular_returns=[], inout_returns=list(*qs))
```

Something like this should fix the mypy error

File: guppylang/prelude/_internal/compiler/quantum.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def __init__(self, opname: str):
     def compile_with_inouts(self, args: list[Wire]) -> CallReturnWires:
         from guppylang.prelude._internal.util import quantum_op
 
-        [q, angle] = args
+        qs, angle = args[:-1], args[-1]
```
Comment (mark-koch) [inline guppylang/prelude/_internal/compiler/quantum.py:73]:
```suggestion
        [*qs, angle] = args
```

File: guppylang/prelude/quantum.py [hunk 3]
```diff
@@ -35,6 +36,10 @@ def h(q: qubit) -> None: ...
 def cz(control: qubit, target: qubit) -> None: ...
 
 
+@guppy.hugr_op(quantum_op("CY"))
+def cy(control: qubit, target: qubit) -> None: ...
```
Comment (mark-koch) [inline guppylang/prelude/quantum.py:40]: Can you also add these functions to the `quantum_functional` module?

File: guppylang/prelude/_internal/compiler/quantum.py (hunk 1 above)
Comment (jake-arkinstall) [reply guppylang/prelude/_internal/compiler/quantum.py]: This gets me:
```
>       return CallReturnWires(regular_returns=[], inout_returns=list(*qs))
E       TypeError: 'OutPort' object is not iterable
```

Note I am assigning qs to the return of self.builder.add_op - I probably need advice on that.

Comment (mark-koch) [reply guppylang/prelude/_internal/compiler/quantum.py]: Oops I think it should be `list(qs)` then

**Pull request merged**